### PR TITLE
Upgrade to Bootstrap 5

### PR DESCRIPTION
**Pull request by mdo**
- Updates dependencies as needed, dropping jQuery and moving to Popper 2
- Updates the Bootstrap import stack
- Moves from modal in example to offcanvas, with a custom CSS adjustment
- Updates JS includes to use offcanvas and dropdown for new demo, including some quick guidance on including necessary DOM scripts (this is probably wrong, but it works)
- Reflect changes in the readme

TODO:

- [x] Fix offcanvas backdrop missing
- [ ] ~~Switch to our bundle JS file and remove popper and custom imports?~~ @mdo: No, I'd like to keep the specific options
- [ ] Move postcss config to postcss.config.js?

### CODE REVIEW

**Comment by alundiak** [comment]
@mdo great that v5 finally released, so one more step is here :) I see `node-sass` removed, as @bardware mentioned in #35  - now it will be good, it must be good.

@mdo will u have time to finalize this PR?

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
   "homepage": "https://github.com/twbs/bootstrap-npm-starter#readme",
   "scripts": {
     "build": "npm run css",
-    "css-compile": "node-sass --include-path node_modules --output-style compressed --source-map true --source-map-contents true --precision 6 scss -o assets/css/",
+    "css-compile": "sass --style expanded --source-map --embed-sources --no-error-css --load-path=node_modules scss/:assets/css/",
     "css-lint": "stylelint scss/",
     "css-prefix": "postcss --replace assets/css/starter.css --use autoprefixer --map",
     "css-purge": "purgecss --keyframes --css assets/css/starter.css --content index.html \"node_modules/bootstrap/js/dist/{util,modal}.js\" --output assets/css/",
```
Comment (XhmikosR) [inline package.json:21]: @mdo We need to adapt this script with the new files.

IMHO we should import the bundle file. This approach is for advanced users.

File: package.json [hunk 2]
```diff
@@ -34,21 +34,21 @@
     "template"
   ],
   "dependencies": {
-    "bootstrap": "^4.6.0",
-    "bootstrap-icons": "^1.4.0",
-    "jquery": "^3.6.0",
-    "popper.js": "^1.16.1"
+    "bootstrap": "^5.0.0",
```
Comment (XhmikosR) [inline package.json:37]: 5.0.1 and icons 1.5.0

File: package.json [hunk 3]
```diff
@@ -34,21 +34,21 @@
...
+    "bootstrap-icons": "^1.4.1",
+    "@popperjs/core": "^2.9.2"
```
Comment (XhmikosR) [inline package.json:39]: This should be moved to the top.

**Comment by XhmikosR** [comment]
Left some comments, I'll try to have a look and fix some of the things later myself.

File: package.json (hunk 1 above)
Comment (XhmikosR) [reply package.json:21]: Also note that if we go with our bundle file, we don't need the separate popper dependency.

File: package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   "homepage": "https://github.com/twbs/bootstrap-npm-starter#readme",
   "scripts": {
     "build": "npm run css",
-    "css-compile": "node-sass --include-path node_modules --output-style compressed --source-map true --source-map-contents true --precision 6 scss -o assets/css/",
+    "css-compile": "sass --style expanded --source-map --embed-sources --no-error-css --load-path=node_modules scss/:assets/css/",
```
Comment (XhmikosR) [inline package.json:18]: We need to change the style to compressed.

File: scss/starter.scss [hunk 5]
```diff
@@ -68,7 +68,7 @@ $success: #7952b3;
 // @import "bootstrap/scss/list-group";
 @import "bootstrap/scss/close";
 // @import "bootstrap/scss/toasts";
-// @import "bootstrap/scss/modal"; // Requires transitions
+@import "bootstrap/scss/modal"; // Requires transitions
```
Comment (XhmikosR) [inline scss/starter.scss]: We might need to communicate this in our docs if we haven't already...

File: package.json [hunk 6]
```diff
@@ -45,7 +45,7 @@
     "postcss": "^8.3.0",
     "postcss-cli": "^8.3.1",
     "purgecss": "^4.0.3",
-    "sass": "1.32.13",
+    "sass": "1.34.0",
```
Comment (XhmikosR) [inline package.json:48]: @mdo I kept this back on purpose

Comment (XhmikosR) [reply package.json:48]: We need to fix the Sass warnings on our side in core first

File: scss/starter.scss [hunk 7]
```diff
@@ -28,37 +28,40 @@
...
+@import "bootstrap/scss/mixins";
+@import "bootstrap/scss/utilities";
+@import "bootstrap/scss/root";
+@import "bootstrap/scss/reboot";
 
-@import "bootstrap/scss/root"; // Required
-@import "bootstrap/scss/reboot"; // Required
 @import "bootstrap/scss/type";
 // @import "bootstrap/scss/images";
 // @import "bootstrap/scss/code";
```
Comment (marcsoler) [inline scss/starter.scss:49]: Even though it's commented out, `bootstrap/scss/code` isn't a valid path

File: scss/starter.scss [hunk 8]
```diff
@@ -28,37 +28,40 @@
...
 // @import "bootstrap/scss/navbar"; // Requires nav
 // @import "bootstrap/scss/card";
 // @import "bootstrap/scss/breadcrumb";
+// @import "bootstrap/scss/accordion";
 // @import "bootstrap/scss/pagination";
 // @import "bootstrap/scss/badge";
-// @import "bootstrap/scss/jumbotron";
 // @import "bootstrap/scss/alert";
 // @import "bootstrap/scss/progress";
 // @import "bootstrap/scss/media";
```
Comment (marcsoler) [inline scss/starter.scss:67]: Even though it's commented out, `bootstrap/scss/media` isn't a valid path

**Comment by mdo** [comment]
@XhmikosR Should we roll in some Webpack or something else for the JS? I've spiked out tries at this, but haven't been happy with what I ended up with. I think we could improve our Webpack docs too in the process if we go that direction.

File: package.json [hunk 9]
```diff
@@ -34,21 +34,20 @@
     "template"
   ],
   "dependencies": {
-    "bootstrap": "^4.6.0",
-    "bootstrap-icons": "^1.4.0",
-    "jquery": "^3.6.0",
-    "popper.js": "^1.16.1"
+    "@popperjs/core": "^2.9.3",
+    "bootstrap": "^5.1.0",
```
Comment (vanillajonathan) [inline package.json:38]: 5.13

**Comment by vanillajonathan** [comment]
@mdo How come you closed this?

**Comment by mdo** [comment]
@vanillajonathan I forgot I had some open PRs here, so I cleaned house today. Everything for v5 examples and starters is at https://github.com/twbs/examples. Lots more places to start, including better starter templates and more complex JS framework setups. All our effort will go into that instead of this repo, which I'll now leave at v4 only.

**Pull request closed without merging**